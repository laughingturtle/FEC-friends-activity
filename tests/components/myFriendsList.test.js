import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
//import 'jsdom-global/register';
import FriendsApp from '../../client/components/app.jsx';
import SongList from '../../client/components/app.jsx';
import SongListEntry from '../../client/components/songList.jsx';
import { mount } from 'enzyme';

describe('<FriendsApp />', () => {
  describe('render()', () => {
  test('renders the component', () => {
    const wrapper = shallow(<FriendsApp />);

    expect(toJson(wrapper)).toMatchSnapshot();
    })
  })
});

describe('<SongList />', () => {
  describe('render()', () => {
  test('renders the component', () => {
    const wrapper = shallow(<SongList />);
  
    expect(toJson(wrapper)).toMatchSnapshot();
    })
  })
});

describe('<SongListEntry />', () => {
  describe('render()', () => {
  test('renders the component', () => {
    const wrapper = shallow(<SongListEntry />);
  
    expect(toJson(wrapper)).toMatchSnapshot();
    })
  })
});

it('ensure to have only three children (control element)', () => {  
  expect(SongListComponent.findWhere(node => node.key() === 'songs').length).toEqual(1);
});

const SongListComponent = mount(<SongList><div /></SongList>);
it('check children prop type', () => {  
      expect(SongListComponent.props().children).toBeObject();
});


