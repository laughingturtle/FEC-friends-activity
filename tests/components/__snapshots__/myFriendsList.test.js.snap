// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FriendsApp /> render() renders the component 1`] = `
<div>
  <h4>
    Friends Activity
  </h4>
  <div
    className=""
  >
    <SongList
      handleSongClick={[Function]}
      songs={
        Array [
          Object {
            "artist": "mr whatsit",
            "firstname": "willy",
            "lastname": "walrus",
            "song_id": 23,
            "title": "bing bang bop",
          },
          Object {
            "artist": "Blim blam",
            "firstname": "Sooch",
            "lastname": "Arechi",
            "song_id": 5,
            "title": "bloot bloot",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`<SongList /> render() renders the component 1`] = `
<div>
  <h4>
    Friends Activity
  </h4>
  <div
    className=""
  >
    <SongList
      handleSongClick={[Function]}
      songs={
        Array [
          Object {
            "artist": "mr whatsit",
            "firstname": "willy",
            "lastname": "walrus",
            "song_id": 23,
            "title": "bing bang bop",
          },
          Object {
            "artist": "Blim blam",
            "firstname": "Sooch",
            "lastname": "Arechi",
            "song_id": 5,
            "title": "bloot bloot",
          },
        ]
      }
    />
  </div>
</div>
`;
